e2e_tests/tests/fixtures/core_api/searcher.py

```diff
@@ -0,0 +1,606 @@
+import argparse
+import dataclasses
+import json
+import logging
+import pickle
+import random
+import sys
+import uuid
+from pathlib import Path
+from typing import Dict, List, Optional, Set, Tuple
+
+from urllib3.connectionpool import HTTPConnectionPool, MaxRetryError
+
+import determined as det
+from determined.common import util
+from determined.searcher.core_search_runner import CoreSearchRunner
+from determined.searcher.search_method import (
+    Close,
+    Create,
+    ExitedReason,
+    Operation,
+    SearchMethod,
+    Shutdown,
+    ValidateAfter,
+)
+
+
+class RandomSearchMethod(SearchMethod):
```

Comment:
The only difference between RandomSearchMethod and AshaSearchMethod here vs in test_custom_searcher.py is that here they accept list of exceptions. will refactor it later today.